pep-9999.rst

```diff
@@ -0,0 +1,1037 @@
+importable by the software in their base system.
+
+While this continues to carry the risk of installing a newer
+version of a dependency than the operating system expects or
+otherwise negatively changing the behavior of an
+application, it does not need to carry the risk of removing
+files from the operating system. A tool like pip should be
+able to install packages in some directory on the default
+``sys.path``, if specifically requested, without deleting
+files owned by the system's package manager.
+
+Therefore, this PEP proposes two things.
+
+First, it proposes **a way for distributors of a Python
+interpreter to mark that interpreter as having its packages
+managed by means external to Python**, such that
+Python-specific tools like pip should not change the
+installed packages in the interpreter's global ``sys.path``
+in any way (add, upgrade/downgrade, or remove) unless
+specifically overridden.  It also provides a means for the
+distributor to indicate how to use a virtual environment as
+an alternative.
+
+This is an opt-in mechanism: by default, the Python
+interpreter compiled from upstream sources will not be so
+marked, and so running ``pip install`` with a self-compiled
+interpreter, or with a distribution that has not explicitly
+marked its interpreter, will work as it always has worked.
+
+Second, it sets the rule that when installing packages to an
+interpreter's global context (either to an unmarked
+interpreter, or if overriding the marking),
+**Python-specific package managers should modify or delete
+files only within the directories of the ``sysconfig``
+scheme in which they would create files**. This permits a
+distributor of a Python interpreter to set up two
+directories, one for its own managed packages, and one for
+unmanaged packages installed by the end user, and ensure
+that installing unmanaged packages will not delete (or
+overwrite) files owned by the external package manager.
+
+Rationale
+=========
+
+As described in detail in the next section, the first
+behavior change involves creating a marker file named
+``EXTERNALLY-MANAGED``, whose presence indicates that
+non-virtual-environment package installations are managed by
+some means external to Python, such as a distribution's
+package manager. This file is specified
+to live in the ``stdlib`` directory in the default
+``sysconfig`` scheme, which marks the interpreter /
+installation as a whole, not a particular location on
+``sys.path``. The reason for this is that, as identified
+above, there are two related problems that risk breaking an
+externally-managed Python: you can install an incompatible
+new version of a package system-wide (e.g., with ``sudo pip
+install``), and you can install one locally, but in a
+location that is loaded by the standard Python command
+(e.g., with ``pip install --user``). If the marker file were
+in the system-wide ``site-packages`` directory, it would not
+clearly apply to the second case. The `Rejected Ideas`_
```

Comment:
```suggestion
clearly apply to the second case. The `Alternatives`_
```

might be a bit softer, particularly if anyone ends up arguing that we go with those solutions... :)

Reply:
"Rejected Ideas" is the name of a standard PEP section, see https://www.python.org/dev/peps/pep-0001/#what-belongs-in-a-successful-pep . I'm not opposed to changing it, but that's why I defaulted to it.

Reply:
That's a shame. I wouldn't want changing this to hurt the PEP's chances.

Reply:
Thinking harder, since it's got the "System Python" subsection, we definitely don't want to call it "Rejected Ideas" because we're emphatically taking no position on it. So, amendment is friendly.